### ISSUE TITLE
out:pbf

### ISSUE THREAD

**Issue by DaveF63** [tyrasd/overpass-turbo#273]
Using the example: http://overpass-turbo.eu/s/jpK I'm unable to get it to work.

Could someone clarify what is meant by 'click on Export -> raw data directly from Overpass API'?

I'm unsure what to click on. Isn't the API command line operated? In Turbo I can only click Export>raw after it's run. By that time it appears I've already obtained 'garbled output'.

Step by Step:
Run example
Obtain garble under 'Data' tab:

![gobbledygook](https://cloud.githubusercontent.com/assets/9528328/20189596/d027c0a6-a775-11e6-8593-df23e0683e7d.JPG)

Click Export>raw data & obtain this:

```
{
  "elements": []
}

```

**Comment by tyrasd** [comment]
> Click Export>raw data & obtain this: […]

that's a bug.

**Comment by mmd-osm** [comment]
It sounds like you're clicking on the wrong link. In the Export popup, there are a two links:

* raw data
* raw data directly from Overpass API

You should click the **second** one, yielding the following download link: [link](http://dev.overpass-api.de/test753/interpreter?data=%5Bout%3Apbf%5D%3B%20%20%20%20%20%20%20%0A%0A%28node%5B%22xmas%3Afeature%22%5D%3B%0A%20way%5B%22xmas%3Afeature%22%5D%3B%0A%20rel%5B%22xmas%3Afeature%22%5D%3B%0A%29%3B%0A%28._%3B%3E%3B%29%3B%0A%20out%20meta%20%3B%0A%0A%0A%0A)

Please keep in mind that this is a prototype only, it is not yet available in the official release yet.

**Comment by tyrasd** [comment]
@mmd-osm right, but it should still be possible to download the data from overpass-turbo via `export -> raw data` once it's loaded in.

It turns out to be relatively tricky to fix, though. The main issue is that the code still builds directly on top of some jQuery features, but jQuery's *ajax* utility doesn't allow one to fetch binary data. It's probably best to replace that with a raw `XMLHttpRequest` (or `fetch`) based solution, but that would require a bit of work.

For the time being, I've created a prototype on the *pbf* branch (deployed at http://overpass-turbo.eu/pbf/) which can **only** parse pbf output. Here's the [example](http://overpass-turbo.eu/pbf/?Q=%5Bout%3Apbf%5D%3B%20%20%20%20%20%20%20%0A%0A(node%5B%22xmas%3Afeature%22%5D%3B%0A%20way%5B%22xmas%3Afeature%22%5D%3B%0A%20rel%5B%22xmas%3Afeature%22%5D%3B%0A)%3B%0A(._%3B%3E%3B)%3B%0A%20out%20meta%20%3B%0A%0A%0A%0A%7B%7Bdata%3Aoverpass%2Cserver%3Dhttp%3A%2F%2Fdev.overpass-api.de%2Ftest753%2F%7D%7D&C=16.46769;-6.15234;2&R) from above.

**Comment by DaveF63** [comment]
> It sounds like you're clicking on the wrong link. 

Self #facepalm. Apologies for not reading accurately.

**Comment by VitalyVaryvdin** [comment]
Sorry for bumping this issue
Are there any updates?
